Overhaul and upgrade of EnemyBar

AkadenTK opened the pull request:
I wanted more features than just a target HPP bar. New features include:
- casting/performing action tracking (graphical)
- crowd control display (if a target/subtarget/etc is bound, slept, petrified, terrorized)
- distance tracking
- rudimentary target-of-target display (essentially, hate tracking, but not as good)
- aggro bars (see: ffxiv) to display health bars and all of the above for enemies that have targeted you or party members.
- focus target bar (see: ffxiv) to display health bar and all of the above for a specified target (player or npc) whether they are targeted or not. 

There are all new commands to control all these settings, as well as specify a focus target. There's a setup mode that allows target bar and aggro frame dragging. Hold control while dragging to "snap" to a grid.

The default modes for all of the above are off. I don't want to disrupt what everyone when they log in for the first time after this change. Settings upgrade automatically, preserving their settings from the old version.

z16 commented on addons/enemybar/bars.lua:
```diff
@@ -0,0 +1,284 @@
+	o.tstat:destroy()
+end
+
+-- Base class method printArea
```
What is `printArea`?

AkadenTK replied:
Something I copy-pasted. Ended up ditching the OOP style because I couldn't figure it out in lua.

z16 commented on addons/enemybar/bars.lua:
```diff
@@ -0,0 +1,284 @@
+
+-- Base class method new
+
+function bars.new(o, bar_settings)
```
You only ever call it with `nil` as `o`. If this is meant as an addon-specific library I'd remove the parameter completely, as it would also fit with the `new` moniker.

z16 commented on addons/enemybar/bars.lua:
```diff
@@ -0,0 +1,284 @@
+	o.atar:destroy()
+	o.ttext:destroy()
+	o.dtext:destroy()
+	o.tstat:destroy()
```
Can't say I'm a big fan of these names :D Would appreciate if you would name them something more specific as it would ease the review process, since it would be more indicative of what each piece is meant to do.

AkadenTK replied:
I agree, I had them short for easy typing, but always planned on changing them. things slip the mind :P

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+clean_actions_tick = clean_actions_delay
+
+function looking_at(a, b)
+	local h = a.facing % math.pi
```
You're mixing different indentation styles throughout the file. Please commit to 4 spaces.

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+	return math.abs(h-h2) < 0.15
+end
+
+function is_party_member_or_pet(mob)
```
This function is rather slow (through little fault of your own). Checking on every action for every target, having to fetch entity objects through the already slow API (and occasionally more than once) and comparing many values for each entity on every action packet may be fast on your PC, but I would not be surprised if this would slow down other player's computers significantly.

There are a number of ways around that, most helpfully though, caching and removing some of the unnecessary checks. You can assume that any function in `windower.ffxi.` runs very poorly, especially `windower.ffxi.get_items`, `windower.ffxi.get_party` and `windower.ffxi.get_player`, two of which you use here. The player ID could be cached throughout the addon for example, as it only changes during logout/login (at which point you can update it).

You could also do some pre-filtering of the ID. For it to be a player ID it must be `id < 0x01000000`, and for it to be a pet ID it must be `id > 0x01000000 && id % 0x1000 > 0x700`. If it is neither, the function may return `false` safely.

But here you can use caching as well, and it should be pretty simple, since you only need to capture the party update packet (0x0DD) as well as the pet info packet (0x067) and build a simple dictionary from there. That would probably be the best solution for the given effort (not much).

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+		track_debuffs(ai)
+	elseif 0x029 then
+    	local message_id = data:unpack('H',0x19)
+    	if not message_id then return end
```
This check can be removed, it never fails (you have a habit of overchecking things throughout the addon, such as checking for `p` above, and a few more times below, I will try to mention it if I think of it).

AkadenTK replied:
Actually, I added this in testing. It was actually failing sometimes without that check. The packet library may have been returning a nil packet because it didn't read properly.

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+	elseif 0x029 then
+    	local message_id = data:unpack('H',0x19)
+    	if not message_id then return end
+    	message_id = message_id%32768
```
I'd use `0x8000` here, seems like less of a magic number.

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+    	local target_id = data:unpack('I',0x09)
+		if S{204,206}:contains(message_id) then
+			-- wears off message.
+			local target = windower.ffxi.get_mob_by_id(target_id)
```
You get the `target` object here, but you only use it for its `.id` field, which you already have (in the `target_id` variable). `windower.ffxi.get_mob_by_id` is not the slowest function in our API, but it certainly is not fast, it should be avoided if not necessary (so much so that I seriously recommend caching its value when it is not hard to implement).

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+end
+
+function track_enmity(ai)
+	local actor = windower.ffxi.get_mob_by_id(ai.actor_id)
```
As I mentioned before, `windower.ffxi.get_mob_by_id` is a slow function, and you call it on the actor and targets in this function, and below in `track_action` and `track_debuffs`. I would call it once in the calling function, and then pass those mob objects as arguments. Also allows you to remove more `if` checks in here, if either of them is `nil` in the calling function you can just return.

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+	for i,t in ipairs(ai.targets) do
+		local target = windower.ffxi.get_mob_by_id(t.id)
+		if target then
+			local pc = nil
```
Just fyi, the ` = nil` is not necessary here and can be removed. You can declare locals with just `local foo` and it will have `nil` as its default value.

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+	end
+
+	-- if the category is not casting magic, jas, items or ws, don't bother.
+	if not S{8,4,7,11,3,6,9,5}:contains(ai.category) then return end
```
This whole `S{1,2,3}:contains` thing has gotten a bit out of hand, and having started it I take full responsibility for it. This is also "slow", although I admit it is not enough to matter much in the end, it just bugs me personally (I would not mind it at all if you left it in). The problem with this is that it constructs the set each time the function is called. It constructs a new table with the values `{8, 4, 7, 11, 3, 6, 9, 5}` and then passes it to the `S` function, which copies all those values into a new table, which is then checked for with `:contains`.

Like I said, it's not a great optimization, but if you care for that kind of thing, you can define these fixed number sets outside of the function and just reuse them:
```lua
local ma_ja_item_ws = S{8, 4, 7, 11, 3, 6, 9, 5}
function track_actions(ai)
    --[[ ... ]]

    if not ma_ja_item_ws:contains(ai.category) then
        return
    end

    --[[ ... ]]
end
```

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+	elseif S{5,9}:contains(ai.category) then
+		action_map = res.items[action_id]
+	else
+		if actor.is_npc then
```
You can combine this and the previous line into an `elseif`. Same for the check below.

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+end
+
+function did_overwrite(target, new, t)
+    if not tracked_debuff[target.id] then
```
This check can never be `true`. You check for that already in the calling function, and if `true` you set an empty table as the value, which will make this check always evaluate to `false`.

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+        local old = res.spells[tracked.spell].overwrites or {}
+        
+        -- Check if there isn't a higher priority debuff active
+        if table.length(old) > 0 then
```
Can remove this check. If the length is `0` the loop will immediately break and you get the same effect. Same for the `if` check below (line 213)

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+end
+
+function check_conflicting_debuffs(ai)
+	local actor = windower.ffxi.get_mob_by_id(ai.actor_id)
```
Another function that calls `windower.ffxi.get_mob_by_id` which could get the entity via an argument instead.

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+				-- it was inactive, but now it's doing!
+				tracked_debuff[actor.id][id] = nil
+			end
+			if math.abs(debuff.pos.x-actor.x)>0.5 or math.abs(debuff.pos.y-actor.y)>0.5 then
```
This and the previous `if` check could be joined with an `or`, but this may actually be better for readability... just pointing it out.

z16 commented on addons/enemybar/actionTracking.lua:
```diff
@@ -0,0 +1,299 @@
+
+	for id,enmity in pairs(tracked_enmity) do
+		if time - enmity.time > 3 then
+			local mob = windower.ffxi.get_mob_by_id(enmity.mob.id)
```
You call that function a few more times in here. These may be new IDs, so not sure if there's a way around it, just pointing it out. I won't make a mention of this again, just search for those occurrences and see what you find that you can optimize.

z16 commented on addons/enemybar/bars.lua:
```diff
@@ -0,0 +1,284 @@
+end
+
+function bars.destroy(o)
+	if not o then return end
```
You check for the existence of `o`, despite it never being `nil`, and also checking again at the call site. You can pretty much remove every one of these `if` checks in this module. It is good practice to do this for public facing libraries to validate unknown input (although I personally do not always agree with that either), but for internal libraries you always control the input, so it is not necessary.

AkadenTK replied:
It actually is called when o could be nil. it's an edge case though. I put it in here to clean up the code in enemybar.lua, since it's used in more places.

z16 commented on addons/enemybar/enemybar.lua:
```diff
@@ -24,31 +24,524 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-require('subtargetBar')
+table = require('table')
+require 'bars'
+require 'actionTracking'
```
If you care for my feelings at all, please surround the required module names with parentheses :(

z16 commented on addons/enemybar/enemybar.lua:
```diff
@@ -24,31 +24,524 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+    end
+    settings = config.load(defaults)
+
+    initialize_bars()
```
The `config` library has a feature you can use for this:
```lua
config.register(settings, initialize_bars)
```

This will execute `initialize_bars` every time the settings are reloaded (i.e. once initially at the start, but also when you relog into a different character, with different settings. You can also manually trigger it with `config.reload(settings)`, as you could do above in the `set_setting` function.

z16 commented on addons/enemybar/enemybar.lua:
```diff
@@ -24,31 +24,524 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+            local action = tracked_actions[target.id]
+            if action and not action.complete then
+                    bars.update_action(bar, action.ability.en, '')
```
Use `.name` instead of `.en` (and use `.english` over `.en` in any case). `.name` will resolve to `.english` on NA clients and to `.japanese` on JP clients.

z16 commented on addons/enemybar/enemybar.lua:
```diff
@@ -24,31 +24,524 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+            worser_match = mob
+        end
+    end
+    return worse_match and worse_match or worser_match
```
This can be just `return worse_match or worser_match`.

z16 commented on addons/enemybar/enemybar.lua:
```diff
@@ -24,31 +24,524 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+end)
+
+if windower.ffxi.get_info().logged_in then
+    player_id = windower.ffxi.get_player().id
```
Here you already do cache the player ID as I suggested above. You can make use of that.

z16 commented on addons/enemybar/enemybar.lua:
```diff
@@ -24,31 +24,524 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+end
+
+function load_settings()
+    defaults = {}
```
The `config` library has a feature where it reloads settings based on a new character itself. You do not need to define this function. Just define the defaults globally and do `local settings = config.load(defaults)`. After a logout/login this object will already contain updated values based on the settings of the new character. See also my note below about `config.register`.

z16 commented on addons/enemybar/enemybar.lua:
```diff
@@ -24,31 +24,524 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+if windower.ffxi.get_info().logged_in then
+    player_id = windower.ffxi.get_player().id
+    load_settings()
```
You should not need to call this, either here or in the `login`/`logout` events. See my comment above at `load_defaults`.

z16 commented on addons/enemybar/enemybar.lua:
```diff
@@ -24,31 +24,524 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+if windower.ffxi.get_info().logged_in then
+    player_id = windower.ffxi.get_player().id
+    load_settings()
+end
```
There is a new line missing at the end of the file!

z16 reviewed (changes requested):
While none of the issues are crucial, I do suggest changes mostly with respect to the performance problem and the deeply nested, repeated calls of `windower.ffxi.get_mob_by_id`. Any `windower.ffxi.get_mob_by_X` function is slow because it has to copy the entire entity from its native data structure to a Lua table, and push dozens of values. Of all of them `windower.ffxi.get_mob_by_id` is often the worst (except for `windower.ffxi.get_mob_by_name`), since it may have to search the entire mob array for the ID (except in certain cases where it can infer the index from the ID, which leads to a direct access, but still makes it just as bad as all the other mob retrieval functions).

Also you should adjust the `.en` thing to `.name` and fix the indentation formatting. And you'd save yourself some effort if you used the `config` library to its full effect (i.e. with its auto-reloading and `config.register`).

magus committed:
areas.Cities +Rabao

z16 committed:
Merge pull request #1746 from magus/patch-1

Add Rabao to Cities set

z16 committed:
Merge pull request #1747 from Windower/dev

Dev > Live merge

Commit:
Refactoring TreasurePool

Refactored to use packets instead of memory

Commit:
fixed formatting

fixed formatting

Commit:
Update TreasurePool.lua

z16 committed:
Merge pull request #1749 from KenshiDRK/dev

Refactoring TreasurePool

Chiaia committed:
Umbra and Rad SC / Fixes Scoreboard

Chiaia committed:
Umbra and Rad SC / Fixes Scoreboard

Chiaia committed:
Slips 28 and Missing other tiems

All of Slip 28 Added
Missing stuff on Slip 22 from Sep 2018 to May 2019 Updated added. (August stuff was added already.)
Missing stuff on Slip 23 from Aug 2018 to May 2019 added.

Chiaia committed:
Slip 28

Chiaia committed:
version bump

Chiaia committed:
Slip 28

z16 committed:
Merge pull request #1751 from Chiaia/dev

Slip 28 Updates

dkolb committed:
Add /boomhelix commands; fix lua error

Now supports SCH Helix Commands.

Also, the conditionals checking the length of the spell_tables entry were out of order, so you could try to get the length of nil instead. Now the or conditional short circuits when the spell_table entry is nil.

General cleanup of code to reuse current_spell_table instead of indexing into spell_tables repeatedly.

Cleaned up repitition in spell casting commands.

dkolb committed:
Style Fixes

Replace tabs witth 4 spaces.

z16 committed:
Merge pull request #1752 from dkolb/dev

EasyNuke: Add /boomhelix commands; fix lua error

Commit:
fix to allow on screen dmg

fix to allow on screen dmg

Commit:
Update battlemod.lua

Commit:
copyright

z16 committed:
Merge pull request #1753 from KenshiDRK/dev

Fix to allow on screen dmg

Commit:
Added Delete function to AzureSets

Commit:
Tabs -> Spaces

Commit:
Version 1.23

Commit:
Damn you MSVS

z16 committed:
Merge pull request #1754 from DYDR/AzureSetsDelete

Added Delete function to AzureSets

Commit:
Fix to allow on screen numbers without simplify and in jp client

Fix to allow on screen numbers when simplify is set to false and in jp client.

Commit:
Some more fixing

Some more fixing

z16 committed:
Merge pull request #1755 from KenshiDRK/dev

Fix to allow on screen numbers without simplify and in jp client

Commit:
[battlemod] Adding Steal/Despoil messages to simplify

Adding Steal/Despoil messages to simplify and added one more blocking chat mode.

Commit:
Added Larceny message

Added Larceny message

z16 committed:
Merge pull request #1756 from KenshiDRK/dev

[Battlemod] Adding Steal/Despoil messages to simplify

z16 committed:
Merged live > dev

z16 committed:
Merge pull request #1757 from Windower/dev

Dev > Live merge

Chiaia committed:
Relic Aug changes for WHM and BLM

Doesn't break anyone's GS files with the way GS handles a single copy of an item in inventory.

johnarnfield committed:
Fix incoming sparks field

As sparks are no longer capped at 50,000, use an unsigned int for sparks quantity rather than unsigned short.

z16 committed:
Merge pull request #1759 from johnarnfield/dev

Fix incoming sparks field

ibm2431 committed:
Update incoming price response packet from using 'Bag' to 'Type'

ibm2431 committed:
Requested comment changes to price response field values

ibm2431 committed:
Update inventory index function in price response

ibm2431 committed:
Change 'Quantity Sold' in NPC price response to 'Count' to be inline with other fields

ibm2431 committed:
Remove the const from the NPC price response packet

Nifim committed:
Merge pull request #1760 from ibm2431/dev

Update fields for incoming price response packet (0x03D)

mac10688 committed:
Adding chatmodes for ls2. Showing valid chatmodes for //scoreboard help. Making error message a little more helpful for tells.

mac10688 committed:
Using single quotes and removing unnecessary parentheses.

z16 committed:
Merge pull request #1761 from mac10688/dev

Scoreboard update and extra info for help and errors.

z16 committed:
Merge pull request #1762 from Windower/dev

Dev > Live merge

mac10688 committed:
Allow sending target id of sender to other characters with <tid>.

mac10688 committed:
Deleting pointless code

mac10688 committed:
Replacing quotes with tick marks

mac10688 committed:
Adding lstid, for sub targetting.

mac10688 committed:
Turned the readme into markdown and gave more information and examples

mac10688 committed:
Changing headers in readme

mac10688 committed:
Fixing small typo

mac10688 committed:
Making the readme more terse

z16 committed:
Merge pull request #1763 from mac10688/dev

Send addon send mobid to receiving players

z16 committed:
Update readme.md

Acbarakat committed:
Improve Summoner CheckParams

There was an issue where Blood Pact Damage would from Pet: and Avatar: did not add up together. My solution is to make all Blood Pact Damage to be Pet: Blood Pact Damage.

Acbarakat committed:
Improve Summoner CheckParams

There was an issue where Blood Pact Damage would from Pet: and Avatar: did not add up together. My solution is to make all Blood Pact Damage to be Pet: Blood Pact Damage.

Acbarakat committed:
Merge branch 'dev' of https://github.com/Acbarakat/Lua into dev

Acbarakat committed:
Update checkparam.lua

Acbarakat committed:
Update checkparam.lua

Acbarakat committed:
Merge branch 'dev' of https://github.com/Acbarakat/Lua into dev

Acbarakat committed:
Update checkparam.lua

z16 committed:
Merge pull request #1764 from Acbarakat/dev

Improve Summoner CheckParams

Chiaia committed:
Fandom changes

Chiaia committed:
Fandom changes

RubenatorX committed:
Use Correct URL Encoding

RubenatorX committed:
Correct styling and scope

z16 committed:
Merge pull request #1766 from RubenatorX/dev

Use Correct URL Encoding

z16 committed:
Merge pull request #1765 from Windower/dev

Dev > Live merge

RubenatorX committed:
Currencies and Currencies 2 updates

RubenatorX committed:
Currencies I and Currencies II packet updates

RubenatorX committed:
Fix merge

z16 committed:
Merge pull request #1767 from RubenatorX/dev

Currencies I and II packet updates

magus committed:
area.Cities + Chocobo Circuit

z16 committed:
Merge pull request #1768 from magus/patch-2

area.Cities + Chocobo Circuit

z16 committed:
Merge pull request #1769 from Windower/dev

Dev > Live merge

RubenatorX committed:
Bugfix for only 1 word searches, had to refactor to account for gsub returning 2 values

RubenatorX committed:
style compliance

z16 committed:
Merge pull request #1770 from RubenatorX/dev

Bugfix for only 1 word searches

Chiaia committed:
giltracker and invtracker

Chiaia committed:
Merge pull request #1771 from Windower/dev

Dev > Live merge

Acbarakat committed:
Update checkparam.lua

- Add Bladeborn/Flashsteel Earring set
- Add Heartseeker/Dudgeon Earring set
- Add in more relevant paramaters to THF's default

Acbarakat committed:
tabs to space-tabs

z16 committed:
Merge pull request #1772 from Acbarakat/dev

Update checkparam.lua (Earring Sets, THF default)

Acbarakat committed:
Missed a comma

Acbarakat committed:
Add in support to display stat caps

- Add in support (display and color code) the params relative to the know (hard) cap
- Do not display stat params that are converted to other stats

Acbarakat committed:
PDT/MDT should be negative

z16 committed:
Merge pull request #1773 from Acbarakat/dev

 Add in support to display stat caps

Acbarakat committed:
Merge pull request #1 from Windower/dev

Get Latest 7/27

Acbarakat committed:
Add support for the alias "Quick Cast"

Quick Cast is the alias for 'occassionally quickens spellcasting' that caps out at 10%

z16 committed:
Merge pull request #1774 from Acbarakat/dev

Enable "Quick Cast"

funkworkz committed:
Update setbgm.lua

Song names released with the soundtrack FINAL FANTASY XI Gifts from Vana'diel: Songs of Rebirth OST.

Nifim committed:
Merge pull request #1775 from funkworkz/patch-4

Update setbgm.lua

Acbarakat committed:
Update checkparam.lua

- Add "Sneak Attack" to default THF
- Add "Trick Attack" to default THF
- Allow "Song Duration" to be bundled with "Song Effect Duration".

z16 committed:
Merge pull request #1776 from Acbarakat/dev

Update checkparam.lua

Acbarakat committed:
Update checkparam.lua

- Add Singing, Wind, and String skill by default to BRD
- Add in Aoido's Matinee and Legato Dagget values to Song Effect Duration

z16 committed:
Merge pull request #1777 from Acbarakat/dev

Update checkparam.lua

Acbarakat committed:
Bugfix in Integrate Key

- Resolve issue where 'blood pact ab. del. ii' could never be found (on such items as Conveyance Cape) due to the period being stripped from the string.

Acbarakat committed:
Tab to Space

Acbarakat committed:
Bugfix Pet: Haste

I could not confirm that gear "Pet: Haste" could go beyond 25%. Setting cap to 25%.

z16 committed:
Merge pull request #1778 from Acbarakat/dev

Bugfix in Integrate Key

Chiaia committed:
Slip 22 Wind Knife NQ/HQ

Chiaia committed:
Slip 22 Wind Knife NQ/HQ

bwjohnson92 committed:
Update Gearswap to allow for Dispelga

Include Dispelga in the check_spell, similar to Impact, as it's a gear-attained spell.

z16 committed:
Merge pull request #1779 from bwjohnson92/dev

Update Gearswap to allow for Dispelga

posimagi committed:
Fix parry pronoun

posimagi committed:
battlemod: handle debuff application messages

z16 committed:
Merge pull request #1781 from posimagi/debuff-application

battlemod: handle debuff application messages

z16 committed:
Merge pull request #1780 from posimagi/dev

Fix parry pronoun in battlemod

Ivaar committed:
Slips.lua: correct item order for ayanmo/add wardrobes

z16 committed:
Merge pull request #1782 from Ivaar/dev

Slips.lua: correct item order for ayanmo/add wardrobes

z16 committed:
Merged live into dev

Ivaar committed:
Slips.lua: added items/verified slip 22

z16 committed:
Merge pull request #1784 from Ivaar/dev

Slips.lua: added items/verified slip 22

Ivaar committed:
Slips.lua: Add ranine staff to Slip 23

z16 committed:
Merge pull request #1786 from Ivaar/dev

Slips.lua: Add ranine staff to Slip 23

KenshiDRK committed:
Adding Mug messages

Adding mug messages for simplify

KenshiDRK committed:
Update parse_action_packet.lua

z16 committed:
Merge pull request #1788 from KenshiDRK/dev

Adding Mug messages to battlemod

KenshiDRK committed:
Adding  Chainbound AE message to battlemod

Adding the chainbound additional effect message from odin in the master trial fight.

Acbarakat committed:
Update 'indicolure effect duration'

Make sure "indi eff dur" equates properly to indicolure effect duration

z16 committed:
Merge pull request #1789 from KenshiDRK/dev

Adding  Chainbound AE message to battlemod

z16 committed:
Merge pull request #1791 from Acbarakat/dev

Checkparam: Update 'indicolure effect duration'

RubenatorX committed:
EasyNuke.lua add command for AncientMagic (//ez boomam)

I'll be honest, haven't tested it yet but, thought I'd toss it in here in case someone wanted to try it out.

z16 committed:
Merge pull request #1792 from RubenatorX/patch-1

EasyNuke.lua add command for AncientMagic (//ez boomam)

KenshiDRK committed:
Add Spirit Bond message to battlemod

Add Spirit Bond message to battlemod

KenshiDRK committed:
Update battlemod.lua

KenshiDRK committed:
Update battlemod.lua

z16 committed:
Merge pull request #1794 from KenshiDRK/dev

Add Spirit Bond message to battlemod

RubenatorX committed:
EasyNuke fix

got pushed with errors

alphaONE2 committed:
Merge pull request #1796 from RubenatorX/patch-2

EasyNuke fix

Ivaar committed:
Gearswap: Filter enchanted items with duplicate names.

z16 committed:
Merge pull request #1797 from Ivaar/dev

Gearswap: Filter enchanted items with duplicate names.

Ivaar committed:
Slips.lua: added Autumn tree and fixed Sakonji Haidate +2

z16 committed:
Merge pull request #1799 from Ivaar/dev

Slips.lua: added Autumn tree and fixed Sakonji Haidate +2

Acbarakat committed:
Add Pet PDT/MDT

- Add WA for Pet: DT to be  shown as Pet: PDT and Pet: MDT much like regulare DT
- Add Pet: PDT/MDT caps

z16 committed:
Merge pull request #1801 from Acbarakat/dev

Add Pet PDT/MDT

Chiaia committed:
Added weather_intensity and real_weather_intensity

Chiaia committed:
version bump

Chiaia committed:
Added world.weather_intensity and in about world.real_weather

Chiaia committed:
Merge pull request #1802 from Chiaia/dev

Gearswap weather_intensity

Chiaia committed:
Rest of relic changes so far.

Chiaia committed:
New Addon autoenterkey

hits enter when first starting the game so you don't timeout on the warning message if you tab out

Chiaia committed:
Merge pull request #1805 from Chiaia/dev

New Addon autoenterkey

KenshiDRK committed:
[battlemod] Adding puppetmaster messages

Adding puppetmaster messages to simplify on battlemod.

Chiaia committed:
Merge pull request #1809 from KenshiDRK/dev

[battlemod] Adding puppetmaster messages

Chiaia committed:
BD Messages Dec 9th, 2019 Update

Sjshovan committed:
Add BattleStations addon

Chiaia committed:
Update fields.lua

Sjshovan committed:
update installation instructions

Chiaia committed:
Update fields.lua

Chiaia committed:
Kenshi Info

Sjshovan committed:
replace tabs with spaces
remove unnecessary string conversions
remove accidental space character in addons.xml

z16 committed:
Merge pull request #1816 from Ap0gee/dev

Add BattleStations addon

Chiaia committed:
Path Augments for Dyna D/Necks

nekonok committed:
add sub command 'wearable' to 'gs export'

z16 committed:
Merge pull request #1820 from nekonok/feature/gearswap-export-wearable

add sub command 'wearable' to 'gs export'

Ivaar committed:
Slips.lua: January 2020 update

z16 committed:
Merge pull request #1821 from Ivaar/dev

Slips.lua: January 2020 update

xurion committed:
Added EmpyPopTracker addon.

xurion committed:
Added EmpyPopTracker to addons.xml.

xurion committed:
Double quotes to singles.
Fixed issue with ucwords where it would return a list.
Futureproofed owns_items function.
Referencing resources by name.
Removed add_to_chat function and moved chat mode to settings.
Removed unnecessary validation from generate_info function.
Removed BG URL from each data file and inferred URL by NM name.
Corrected indent size in data files.
PR comments addressed.

xurion committed:
Switched to pattern matching.
Reverted indents to 4 spaces.
Commands now handle different cases.

z16 committed:
Merge pull request #1822 from xurion/dev

New addon - EmpyPopTracker

genoxd committed:
Clean up a tab

genoxd committed:
Adding check for message_id 802 to cover the new Siren HP drain BP

z16 committed:
Merge pull request #1829 from NassirAlawar/dev

Handling the new Siren BP that does HP drain

Chiaia committed:
Myhome warp all support

Chiaia committed:
Dimmer

Chiaia committed:
Update extdata.lua

xurion committed:
Added WoC and ADL to trackable NMs.
Adjusted addon to allow tracking of NMs that use regular items as pops.
Allowed tracking of NMs with spaces in their name.

xurion committed:
Reduced generate_info to make it more concise.
https://github.com/Windower/Lua/pull/1834

z16 committed:
Merge pull request #1834 from xurion/dev

Added WoC and ADL to trackable NMs

RubenatorX committed:
Add missing Ionis npc

Nifim committed:
Merge pull request #1835 from RubenatorX/patch-3

Add missing Ionis npc

Nifim committed:
Update bufftime offset

bufftime offset has changed to `572662306`

z16 committed:
Merge pull request #1837 from Windower/update_bufftime_offset

Update bufftime offset

xurion committed:
EmpyPopTracker 2.1.1.
Fixed issue where items would not show green.
Updated addon description.

z16 committed:
Merge pull request #1838 from xurion/dev

EmpyPopTracker fix - items would not show green when obtained

dtw committed:
Add burdentracker to settings

z16 committed:
Merge pull request #1840 from dtw/autocontrol

Add burdentracker to settings

lili-ffxi committed:
Create indinope.lua

lili-ffxi committed:
Create README.md

lili-ffxi committed:
Update addons.xml

lili-ffxi committed:
Update addons.xml

lili-ffxi committed:
Update indinope.lua

lili-ffxi committed:
Update README.md

lili-ffxi committed:
Update indinope.lua

lili-ffxi committed:
Update README.md

lili-ffxi committed:
Update indinope.lua

lili-ffxi committed:
Update indinope.lua

z16 committed:
Merge pull request #1843 from lili-ffxi/dev

IndiNope addon.

xurion committed:
Fixed incorrect Chloris data.
Allowed configurable text colours.

z16 committed:
Merge pull request #1844 from xurion/dev

EmpyPopTracker 2.2.1

danpelope committed:
Trusts - new trust

Add support for Monberaux (added in March 2020 version update)

danpelope committed:
Trusts whitespace

My apologies.

z16 committed:
Merge pull request #1846 from danpelope/trusts-monberaux

Trusts - new trust

Ivaar committed:
Slips.lua: March 2020 update

z16 committed:
Merge pull request #1848 from Ivaar/dev

Slips.lua: March 2020 update

dtw committed:
Change color of recommended action

Just to make it stand out more in the log.

dtw committed:
Capitalize chat msgs

Because OCD

dtw committed:
Add suggested amends

z16 committed:
Merge pull request #1850 from dtw/dtw-boxdestroyer

Dtw boxdestroyer

xurion committed:
EmpyPopTracker 2.2.2.
Fixed issue where text bg colour would not reset.
Added default bg colours to settings.

z16 committed:
Merge pull request #1851 from xurion/dev

EmpyPopTracker 2.2.2

Lowclock committed:
Update fields.lua

Lowclock committed:
Update fields.lua

Nifim committed:
Merge pull request #1852 from Lowclock/dev

Added Unity Rank to fields.lua, and renamed a few duplicated _unknowns.

Lowclock committed:
Corrected labels.

Corrected incoming 0x065 labels for Animation and Rotation.

Chiaia committed:
Merge pull request #1853 from Lowclock/dev

Corrected labels.

Chiaia committed:
Tidying up a bit

Chiaia committed:
Update triggers.lua

Chiaia committed:
fixing varibles

Chiaia committed:
Update organizer.lua

Chiaia committed:
Update boxdestroyer.lua

Chiaia committed:
Update organizer.lua

posimagi committed:
battlemod: Englishier monster level upon check

z16 committed:
Strings: Added string.encode

z16 committed:
Packets: Added encoding mechanism to packets.build

z16 committed:
Packets: Fixed encoded string handling

z16 committed:
Merge pull request #1859 from posimagi/battlemod-check-level

battlemod: Englishier monster level upon check

z16 committed:
Strings: Fixed various encoding issues

tehkrizz committed:
Merge remote-tracking branch 'refs/remotes/Windower/dev'

tehkrizz committed:
Merge branch 'master' of https://github.com/tehkrizz/Lua

tehkrizz committed:
Salvage2 Update

I have restructured parts of Salvage2 so that it uses the new Texts for
boxes, and it now works again.

tehkrizz committed:
DynamisHelper Update

Change textbox method to use Texts, and made other structure adjustments.

tehkrizz committed:
Merge pull request #6 from Windower/dev

Updating local repository

tehkrizz committed:
Revert "DynamisHelper Update"

This reverts commit 4554c09b124c0de841855f8c15d0008d69da8b90.

z16 committed:
Packets: Fixed VLA field injection

z16 committed:
Packets: Fixed bit packed string encoding

z16 committed:
Packets: Fixed bit packed string decoding

z16 committed:
Fields/0x017: Fixed field definitions

tehkrizz committed:
Delete settings.xml

tehkrizz committed:
Delete settings.xml

tehkrizz committed:
Merge branch 'master' of https://github.com/tehkrizz/Lua

tehkrizz committed:
Salvage2 Tabs to Spaces

As requested

z16 committed:
Fields/LS: Fixed bit packed ctype

z16 committed:
Packets: Fixed incorrect null terminator

z16 committed:
Fields/0x017: Fixed message ctype

xurion committed:
EmpyPopTracker 2.3.0 - added colorless soul NMs.

z16 committed:
Merge pull request #1861 from xurion/dev

EmpyPopTracker 2.3.0 - Colorless Soul NMs

tehkrizz committed:
Salvage2 and THTracker parenthesis and tabs

Parenthesis and tabs

z16 committed:
Merge pull request #1860 from tehkrizz/master

Salvage2 Update

xurion committed:
Added NoCampaignMusic addon.

xurion committed:
NCM: Added toggle for notifications.
Fixed issue where addon would perma think campaign was active.
PR comments.

xurion committed:
NCM: Optimised incoming chunk logic.
Removed on/off commands.
Converted settings to CamelCase.
Fixed incorrect asignment of music for solo/party battle on zone update.
PR comments.

z16 committed:
Merge pull request #1862 from xurion/dev

New addon - NoCampaignMusic

AkadenTK committed:
Merge remote-tracking branch 'origin/dev' into dev
